### PR TITLE
Replace C standard library headers

### PR DESCRIPTION
**Pull request by rettinghaus**
This is a small cosmetic change to prefer `cassert` and `cstdlib` over the C standard library headers.
A corresponding commit for libmei has been added to the [open PR](https://github.com/rism-digital/libmei/pull/16).

### CODE REVIEW

**Comment by ahankinson** [comment]
Out of curiosity, what's the advantage?

**Comment by rettinghaus** [comment]
There's no real advantage to us, except usage of C standard library headers was considered deprecated in C++17. Also they are simply a subset of the respective C++ headers. 

And there may be a bit more clarity in using namespaces: 

C++17 standard

**Comment by rettinghaus** [comment]
The [C++20 standard](https://isocpp.org/files/papers/N4860.pdf) states:
> For compatibility with the C standard library, the C++ standard library provides the C headers enumerated in D.9, but their use is deprecated in C++.

**Comment by ahankinson** [comment]
ok, thanks

**Comment by lpugin** [comment]
I think that in fact C++20 has un-deprecated them

https://open-std.org/JTC1/SC22/WG21/docs/papers/2017/p0619r0.html#3.5

File: src/pugi/pugixml.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
-#include <assert.h>
```
Comment (lpugin) [inline src/pugi/pugixml.cpp:22]: You should open a PR on pugixml and not change it here.

Comment (rettinghaus) [reply src/pugi/pugixml.cpp:22]: Sorry. Actually there is one already: https://github.com/zeux/pugixml/pull/350/

**Pull request merged**